llvm/lib/Target/LoongArch/LoongArchInstrInfo.td

```diff
@@ -734,6 +734,7 @@ def ADD_W : ALU_3R<0x00100000>;
 def SUB_W : ALU_3R<0x00110000>;
 def ADDI_W : ALU_2RI12<0x02800000, simm12_addlike>;
 def ALSL_W : ALU_3RI2<0x00040000, uimm2_plus1>;
+let isReMaterializable = 1 in
```

Comment:
For readability, It's better to add surrounding `{ ... }` since this is no blank line between this `def` and the next one.